firebase-common/src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -415,18 +417,25 @@ protected FirebaseApp(Context applicationContext, String name, FirebaseOptions o
     this.name = Preconditions.checkNotEmpty(name);
     this.options = Preconditions.checkNotNull(options);
 
+    FirebaseTrace.pushTrace("Firebase");
```

Comment:
Thinking out loud: inside `ComponentMonitor` you trace with a `try-finally` to make sure we don't leave start/end unbalanced. Would it make sense to introduce nested try-finally blocks here? wdyt?

Reply:
I assumed if something threw in the `FirebaseApp` constructor, we'd have more worries than imbalanced tracing. Avoidance is mostly style concern, messes up git blame with indentation, goes pretty deep, and adds a lot of finally blocks which could impair reading. If you'd prefer I could go for it though.

Reply:
sgtm